contracts/utils/EnumerableSet.sol

```diff
@@ -196,14 +196,16 @@ library EnumerableSet {
         uint256 valueIndex = set._indexes[value];
 
         if (valueIndex != 0) {
-            uint256 index = valueIndex - 1;
-            bytes32 last = set._values[set._values.length - 1];
+            unchecked {
+                uint256 index = valueIndex - 1;
 
-            // move last value to now-vacant index
+                bytes32 last = set._values[set._values.length - 1];
 
-            set._values[index] = last;
-            set._indexes[last] = index + 1;
+                // move last value to now-vacant index
 
+                set._values[index] = last;
+                set._indexes[last] = index + 1;
```

Comment:
I don't agree with the readability argument from the Discord chat.  Better to use the value already stored in a variable.

```suggestion
                set._indexes[last] = valueIndex;
```

Reply:
And with that change, we probably don't need a reference to `index`, because it will only be used once.

Reply:
I don't feel strongly about readability re `set._indexes[last] = valueIndex;` vs `set._indexes[last] = index + 1;`, happy to go either way. 

But overall I do like the idea of removing `index` completely, feel like that makes it easier to follow the logic, without having to memorize 2 index-liked variables.